src/main/webapp/app/exercises/quiz/manage/quiz-exercise-lifecycle-buttons.component.ts

```diff
@@ -0,0 +1,141 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { QuizExercise, QuizMode, QuizStatus } from 'app/entities/quiz/quiz-exercise.model';
+import { QuizExerciseService } from './quiz-exercise.service';
+import { ActionType } from 'app/shared/delete-dialog/delete-dialog.model';
+import { AlertService } from 'app/core/util/alert.service';
+import { faEye, faFileExport, faPlayCircle, faPlus, faSignal, faSort, faStopCircle, faTable, faTimes, faWrench } from '@fortawesome/free-solid-svg-icons';
+import { Subject } from 'rxjs';
+
+@Component({
+    selector: 'jhi-quiz-exercise-lifecycle-buttons',
+    templateUrl: './quiz-exercise-lifecycle-buttons.component.html',
+})
+export class QuizExerciseLifecycleButtonsComponent {
+    protected readonly QuizMode = QuizMode;
+    protected readonly QuizStatus = QuizStatus;
+    protected readonly ActionType = ActionType;
+
+    // Icons
+    faSort = faSort;
+    faPlus = faPlus;
+    faTimes = faTimes;
+    faEye = faEye;
+    faWrench = faWrench;
+    faTable = faTable;
+    faSignal = faSignal;
+    faFileExport = faFileExport;
+    faPlayCircle = faPlayCircle;
+    faStopCircle = faStopCircle;
+
+    protected dialogErrorSource = new Subject<string>();
+    dialogError$ = this.dialogErrorSource.asObservable();
+
+    @Input()
+    quizExercise: QuizExercise;
+
+    @Output()
+    loadOne = new EventEmitter<number>();
+
+    @Output()
+    handleNewQuizExercise = new EventEmitter<QuizExercise>();
+
+    constructor(
+        private quizExerciseService: QuizExerciseService,
+        private alertService: AlertService,
+    ) {}
+
+    /**
+     * Set the quiz open for practice
+     */
+    openForPractice() {
+        this.quizExerciseService.openForPractice(this.quizExercise.id!).subscribe({
+            next: (res: HttpResponse<QuizExercise>) => {
+                this.handleNewQuizExercise.emit(res.body!);
+            },
+            error: (res: HttpErrorResponse) => {
+                this.onError(res);
+                this.loadOne.emit(this.quizExercise.id!);
```

Comment:
I this `this.loadOne.emit(this.quizExercise.id!);` can be put inside `onError` method.

Reply:
ah, yes you are right 👍